### ISSUE TITLE
Crash when no save file present

### ISSUE THREAD

**Issue by Sanqui** [N64N64/LunaRoja#1]
luared crashes after loading a ROM without a corresponding `.sav`.  Simply creating the file fixes this.

```
sanky@suguri:~/romhacking/gb/luared/luared$ love .
Error: /home/sanky/romhacking/gb/luared/luared/src/lua/gameboy.lua:36: savefile is NULL
stack traceback:
	[C]: in function 'error'
	/home/sanky/romhacking/gb/luared/luared/src/lua/gameboy.lua:36: in function 'new'
	/home/sanky/romhacking/gb/luared/luared/src/lua/init.lua:5: in function 'v'
	/home/sanky/romhacking/gb/luared/luared/src/lua/toggler.lua:116: in function 'update'
	/home/sanky/romhacking/gb/luared/luared/src/lua/toggler.lua:12: in function 'render'
	/home/sanky/romhacking/gb/luared/luared/src/lua/init.lua:28: in function 'mode'
	/home/sanky/romhacking/gb/luared/luared/src/lua/init.lua:100: in function 'MAIN_LOOP'
	...y/romhacking/gb/luared/luared/src/lua/plat/love/init.lua:25: in function 'update'
	[string "boot.lua"]:464: in function <[string "boot.lua"]:436>
	[C]: in function 'xpcall'
sanky@suguri:~/romhacking/gb/luared/luared$ touch rom/pokered.sav
sanky@suguri:~/romhacking/gb/luared/luared$ love .
```

**Comment by N64N64** [comment]
The crash actually has to do with being unable to save to the `.sav` file. It checks for that on ROM load.

mGBA has some weird way of opening/closing files (VFile), I'll need to change it to stdio or something.

**Comment by N64N64** [comment]
The crash has to do with being unable to write to the `.sav` file, so saves won't work for you. It checks for that on ROM load, hence the crash.

I think it has to do with Love2D's filesystem sandbox interacting weirdly with mGBA. mGBA has some custom way of opening/closing files (VFile), I'll need to change it to stdio or something.

Weirdly enough, this issue isn't present on Mac. Only present on Windows and Linux.

**Comment by N64N64** [comment]
Nope. I just (naively) assumed that O_CREAT is the same on all platforms. Fixed in f18ed34816975e1d3109cf2ea1043d3ab837dcc1.
